up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,410 @@ SPDX-FileType: DOCUMENTATION
+[#usubscription-default-notification-topic]
+[.specitem,oft-sid="req~usubscribe-change-notification-topic~1",oft-needs="impl,utest"]
+--
+uSubscription service *SHALL* publish change notification messages to the topic
+ `/core.usubscription/3/subscriptions#Update`.
+--
 
-* *MUST* *NOT* be permitted to call `RegisterForNotifications()` for topics they do not produce, i.e. the uEntity sending the request is not equal to the `UEntity` of the request message's `UUri` parameter.
-* Subscription change notifications *MUST* be sent for changes to `SubscriptionState` for any subscriber that is subscribed to the topic
+=== Custom notification topics
 
-=== Dispatchers
+In addition to publishing subscription change notifications on the link:#usubscription-default-notification-topic[default notification channel], uEntities may want to receive subscription change notifications on a topic of their choice. 
 
-Dispatchers are also permitted to register for subscription change notifications to facilitate the multicasting of messages. The mechanics and requirements of the dispatcher and uSubscription communication are platform deployment specific. 
+[.specitem,oft-sid="dsn~usubscribe-register-notifications~1",oft-needs="impl,utest"]
+--
+Clients *MAY* use `RegisterForNotifications()` to receive subscription change notification messages on a custom topic of their choice.
+--
 
-== Event Delivery
+[.specitem,oft-sid="dsn~usubscribe-unregister-notifications~1",oft-needs="impl,utest"]
+--
+Clients *MAY* use `UnregisterForNotifications()` to stop receiving subscription change notification messages on a custom topic.
+--
 
-The uSubscription service also allows to indicate to subscribers how they shall consume published events by means of the `EventDeliveryConfig` returned in the `SubscriptionResponse` message. For example, if the subscriber has to consume from a different topic or from different messaging infrastructure, this message will store the corresponding delivery semantics.
+== Timeout & Retry Logic
 
-NOTE: Delivery semantics (if any) are deployment specific and not covered in this specification
+Subscribe (and unsubscribe) to remote topics are handled by RPC calls between uSubscription services running on the different devices. Given that devices are not always connected to each other, the onus is on uSubscription service to ensure that a command is received in time. Below are the common retry and timeout policies for USubscription service implementations to follow:   
+
+[.specitem,oft-sid="req~usubscribe-remote-max-timeout~1",oft-needs="impl"]
+--
+Remote Subscribe/Unsubscribe requests *MUST* have a maximum timeout of 5 minutes
+--
+
+[.specitem,oft-sid="req~usubscribe-remote-retry-indefinitely~1",oft-needs="impl"]
+--
+Timed-out remote commands *MUST* be retied indefinitely until the business logic behind it no longer requires the command to be sent (ex. the last subscriber calls unsubscribes from a topic that is in state `SUBSCRIPTION_PENDING`). 
```

Comment:
```suggestion
Timed-out remote commands *MUST* be retried indefinitely until the business logic behind it no longer requires the command to be sent (e.g. because the last entity unsubscribed from a topic that is in state `SUBSCRIPTION_PENDING`). 
```